### PR TITLE
Upgrades Participant API - Enable Tests

### PR DESCRIPTION
**Pull request by dylant-da**
Add back tests from https://github.com/digital-asset/daml/pull/18102 now that https://github.com/DACH-NY/canton/pull/16915/ got merged and code dropped to Daml.

### CODE REVIEW

File: daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+import org.scalatest.compatible.Assertion
+
+import scala.io.Source
+import com.daml.lf.data.Ref.PackageId
+
+import com.daml.lf.archive.{DarReader}
+import scala.util.{Success, Failure}
+
+final class UpgradesIT extends AsyncWordSpec with Matchers with Inside with CantonFixture {
+  self: Suite =>
```
Comment (tudor-da) [inline daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala:26]: Why do you need this self-type here?

File: daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+  override lazy val devMode = true;
```
Comment (tudor-da) [inline daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala:27]: Do you want to keep this devMode enabled?

**Review by tudor-da — APPROVED** [review]
lgtm!

File: daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala (hunk 2 above)
Comment (dylant-da) [reply daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala:27]: turning it off fails the test, though I'm not sure why - it should only control `devVersionSupport` in `ParticipantNodeParameters`.

File: daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala (hunk 1 above)
Comment (dylant-da) [reply daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala:26]: removed, it isn't necessary

File: daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala (hunk 2 above)
Comment (tudor-da) [reply daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala:25]: ah ok, this is the devMode for Canton. All good then 👍

**Pull request merged**